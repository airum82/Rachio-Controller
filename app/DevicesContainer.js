import React from 'react';
import Device from './Device';
import { NavLink, Route } from 'react-router-dom';

const DevicesContainer = ({ devices }) => {
  return (
    <div className="devices-container">
      { devices.map((device, index) => {
          return (
            <div className="devices" key={index}>
              <Route exact path='/' render={() => {
                return (
                  <NavLink to={`/${device.id}`} className="device-link">
                    {device.name + ': ' + device.status}
                  </NavLink>
                )
              }} />
              <Route exact path={`/${device.id}`} render={({ history }) => {
                return (
                  <Device
                    id={device.id}
                    name={device.name}
                    zones={device.zones}
                    history={history}
                  />
                )
              }} />
            </div>
          )
        })}
    </div>
  )
}

export default DevicesContainer;