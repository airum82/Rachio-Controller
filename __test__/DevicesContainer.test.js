import React from 'react';
import DevicesContainer from '../app/DevicesContainer';
import { shallow } from 'enzyme';

describe('DevicesContainer', () => {
  it('should match snapshot upon initial render', () => {
    const wrapper = shallow(<DevicesContainer devices={[{}, {}]}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
