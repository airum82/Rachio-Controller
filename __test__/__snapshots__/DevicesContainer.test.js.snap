// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevicesContainer should match snapshot upon initial render 1`] = `
<div
  className="devices-container"
>
  <div
    className="devices"
    key="0"
  >
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/undefined"
      render={[Function]}
    />
  </div>
  <div
    className="devices"
    key="1"
  >
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/undefined"
      render={[Function]}
    />
  </div>
</div>
`;
